### PR TITLE
Added second param to dependency array in useEffect

### PR DESCRIPTION
**Pull request by chenjie-b1**
This PR aims to provide a fix on the issue(#256). 

If the same checkbox is selected twice, then the dropdown loses focus and does not close when you click outside. Please see the video below and the relevant code [sandbox](https://codesandbox.io/s/youthful-tesla-ocohd). Credits to @mkhbragg.

The issue occur because the `focused` variable does not change when i check and uncheck the same box (I did a `console.log(focused)` in the `select-item.tsx` file). This resulted in the useEffect not being run and the onBlur event being triggered on the second click. 

Suggested solution
```
useEffect(() => {
    updateFocus();
  }, [checked, focused]); 
```

By adding a second parameter, `checked` in the array dependency, the useEffect is being trigger either whenever the check or focused changes. This prevents the dropdown item from losing focus and selection with arrow keys will still work fine.

https://user-images.githubusercontent.com/76419896/105120648-9d7f4c00-5b0d-11eb-8ae9-d1d6dc575e16.mp4

### CODE REVIEW

File: src/select-panel/select-item.tsx [hunk 1]
```diff
@@ -46,8 +46,7 @@ const SelectItem = ({
 
   useEffect(() => {
     updateFocus();
-    // eslint-disable-next-line
-  }, [focused]);
+  }, [checked, focused]);
```
Comment (harshzalavadiya) [inline src/select-panel/select-item.tsx:49]: please add `// eslint-disable-next-line` back since it's necessary for lint

**Review by harshzalavadiya — CHANGES_REQUESTED** [review]
@chenjie-b1 Thanks for fixing this issue however one minor change is required please fix this so we can merge this

File: src/select-panel/select-item.tsx (hunk 1 above)
Comment (chenjie-b1) [reply src/select-panel/select-item.tsx:49]: Sure thank you

**Comment by harshzalavadiya** [comment]
released `v3.1.2`

**Pull request merged**